tests/test_api_requests.py

```diff
@@ -310,3 +506,742 @@ async def parse(self, response):
+        (
+            {
+                "meta": {
+                    "zyte_api": {
+                        "browserHtml": True,
+                        "httpResponseHeaders": False,
+                    },
+                },
+            },
+            {},
+            {
+                "browserHtml": True,
+                "httpResponseHeaders": False,
+            },
+            [],
+        ),
+        # METHOD
+        # Request.method is mapped as is.
+        *(
+            (
+                {"method": method},
+                {},
+                {
+                    "httpResponseBody": True,
+                    "httpResponseHeaders": True,
+                    "httpRequestMethod": method,
+                },
+                [],
+            )
+            for method in (
+                "POST",
+                "PUT",
+                "DELETE",
+                "OPTIONS",
+                "TRACE",
+                "PATCH",
+            )
+        ),
+        # Request.method is mapped even for methods that Zyte Data API does not
+        # support.
+        *(
+            (
+                {"method": method},
+                {},
+                {
+                    "httpResponseBody": True,
+                    "httpResponseHeaders": True,
+                    "httpRequestMethod": method,
+                },
+                [],
+            )
+            for method in (
+                "HEAD",
+                "CONNECT",
+                "FOO",
+            )
+        ),
+        # An exception is the default method (GET), which is not mapped.
+        (
+            {"method": "GET"},
+            {},
+            {
+                "httpResponseBody": True,
+                "httpResponseHeaders": True,
+            },
+            [],
+        ),
+        # httpRequestMethod should not be defined through meta.
+        (
+            {
+                "meta": {
+                    "zyte_api": {
+                        "httpRequestMethod": "GET",
+                    },
+                },
+            },
+            {},
+            {
+                "httpResponseBody": True,
+                "httpResponseHeaders": True,
+                "httpRequestMethod": "GET",
+            },
+            ["Use Request.method instead"],
+        ),
+        # If defined through meta, httpRequestMethod takes precedence, warning
+        # about value mismatches.
+        (
+            {
+                "method": "POST",
+                "meta": {
+                    "zyte_api": {
+                        "httpRequestMethod": "PATCH",
+                    },
+                },
+            },
+            {},
+            {
+                "httpResponseBody": True,
+                "httpResponseHeaders": True,
+                "httpRequestMethod": "PATCH",
+            },
+            [
+                "Use Request.method instead",
+                "does not match the Zyte Data API httpRequestMethod parameter",
```

Comment:
To make the warning message more clear, we can mention which value is actually used.

The same is true for the header warnings in the test cases below.